src/azure-cli/azure/cli/command_modules/acs/azuremonitormetrics/dc/defaults.py

```diff
@@ -13,10 +13,10 @@
 # All DC* object names should end only in alpha numeric (after `length` trim)
 # DCE remove underscore from cluster name
 def sanitize_name(name, objtype, length):
+    name = name[0:length]
     length = length - 1
```

Comment:
Seems `length = length -1` is useless after the change

Reply:
Please add some UT

Reply:
@FumingZhang I added unit tests for the function. Thanks!